src/mongo/db/fts/ngram-tokenizer.md

```diff
@@ -0,0 +1,107 @@
+# NGRAM
+
+###1) TOKEN DELIMITER
+Unlike MongoDB native delimiter based tokenizer, NGram use only below 5 characters as tokenizer delimiter.
+```
+0x09 : HORIZONTAL-TAB
+0x0a : LINE-FEED (\n)
+0x0b : VERTICAL-TAB
+0x0d : CARRIAGE-RETURN (\r)
+0x20 : SPACE
+```
+
+In MongoDB native full text search, we should use `phrase search` to search the string which contains "-" character.
+```
+db.collection.find({ $text: { $search: "\"2016-12\"" } })
+```
+
+But using NGram tokenizer, we don't need to use `phrase search` because "-" is not a delimiter in NGram tokenizer.
+```
+db.collection.find({ $text: { $search: "2016-12" } })
+
+```
+
+Exceptionally if you use "-" character at first of search string, then NGram search understand that "-" character as negate sign (like MongoDB native full text search). 
+
+For example, NGram full text search does not regarding "-"(in the middle of string) as either delimiter and negate-sign.
+```
+db.collection.find({ $text: { $search: "6-2" } })
+```
+But below case, NGram full text search will do negate search for string of "6-2".
```

Comment:
I think this functionality should be tested by jstest.

Reply:
I have.
See https://github.com/SunguckLee/percona-server-mongodb/commit/94492a05cd0fff93c010f330c2019e89bc42234c#diff-814e4c43d9234a5754f4f5b16608a904R34

Reply:
I meant testing of negate search. It should be something with leading minus sign, not sure what the correct syntax is:
```
db.collection.find({ $text: { $search: "-6-2" } }) 
```
or
```
db.collection.find({ $text: { $search: "-\"6-2\"" } })
```